src/transformer/descriptor/type/type.ts

```diff
@@ -41,7 +41,20 @@ export function GetType(node: ts.Node, scope: Scope): ts.Node {
       node.typeName
     );
 
-    return GetType(declaration, scope);
+    const type: ts.Node = GetType(declaration, scope);
+
+    // if the type is a reusable and finite (reusable types except for typeAlias)
+    // we can return the original node
+    // so it can be re used and we handle recursion
+
+    if (
```

Comment:
does it mean that in case of an object type it will still give error?

```
  type B = A;

  type A = {
    recursiveProp: B | null;
    aProp: string | null;
  };
  
  createHydratedMock<A>();
```

Reply:
Good point! Maybe I should check if its an object like :)

Reply:
Done :)